clerk/handler_test.go

```diff
@@ -0,0 +1,171 @@
+	result = suite.handler(ctx, msgEventRecord)
+	require.False(t, result.IsOK(), "error record already synced %v", result)
+
+	chainId = "1"
+	id = uint64(2)
+	index = simulation.RandIntBetween(r1, 0, 100)
+	logIndex = uint64(index)
+	msgEventRecord = types.NewMsgEventRecord(
+		hmTypes.BytesToHeimdallAddress(address.Bytes()),
+		txHash,
+		logIndex,
+		id,
+		chainId,
+	)
+	suite.contractCaller.On("GetConfirmedTxReceipt", mock.Anything, txHash.EthHash(), chainParams.TxConfirmationTime).Return(txreceipt, nil)
+	suite.contractCaller.On("DecodeStateSyncedEvent", chainParams.ChainParams.StakingInfoAddress.EthAddress(), txreceipt, msgEventRecord.LogIndex).Return(eventLog, nil)
+	result = suite.handler(ctx, msgEventRecord)
+	require.False(t, result.IsOK(), "error invalid bor chain id %v", result)
+
+	chainId = "15001"
+	id = uint64(3)
+	index = simulation.RandIntBetween(r1, 0, 100)
+	logIndex = uint64(index)
+	msgEventRecord = types.NewMsgEventRecord(
+		hmTypes.BytesToHeimdallAddress(address.Bytes()),
+		txHash,
+		logIndex,
+		id,
+		chainId,
+	)
+	suite.contractCaller.On("GetConfirmedTxReceipt", mock.Anything, txHash.EthHash(), chainParams.TxConfirmationTime).Return(txreceipt, nil)
+	suite.contractCaller.On("DecodeStateSyncedEvent", chainParams.ChainParams.StakingInfoAddress.EthAddress(), txreceipt, msgEventRecord.LogIndex).Return(eventLog, nil)
+	result = suite.handler(ctx, msgEventRecord)
+	require.False(t, result.IsOK(), "error message log id and event log id dont match %v", result)
+
+	chainId = "15001"
+	id = uint64(4)
+	index = simulation.RandIntBetween(r1, 0, 100)
+	logIndex = uint64(index)
+	msgEventRecord = types.NewMsgEventRecord(
+		hmTypes.BytesToHeimdallAddress(address.Bytes()),
+		txHash,
+		logIndex,
+		id,
+		chainId,
+	)
+	eventLog = &statesender.StatesenderStateSynced{
+		Id:              big.NewInt(4),
+		ContractAddress: address,
+	}
+	suite.contractCaller.On("GetConfirmedTxReceipt", mock.Anything, txHash.EthHash(), chainParams.TxConfirmationTime).Return(txreceipt, nil)
+	suite.contractCaller.On("DecodeStateSyncedEvent", chainParams.ChainParams.StakingInfoAddress.EthAddress(), txreceipt, msgEventRecord.LogIndex).Return(nil, nil)
+	result = suite.handler(ctx, msgEventRecord)
+	require.False(t, result.IsOK(), "error DecodeStateSyncedEvent %v", result)
+
+	// chainId = "15001"
+	// id = uint64(5)
+	// index = simulation.RandIntBetween(r1, 0, 100)
+	// logIndex = uint64(index)
+	// msgEventRecord = types.NewMsgEventRecord(
+	// 	hmTypes.BytesToHeimdallAddress(address.Bytes()),
+	// 	txHash,
+	// 	logIndex,
+	// 	id,
+	// 	chainId,
+	// )
+	// eventLog = &statesender.StatesenderStateSynced{
+	// 	Id:              big.NewInt(5),
+	// 	ContractAddress: address,
+	// }
+	// suite.contractCaller.On("GetConfirmedTxReceipt", mock.Anything, txHash.EthHash(), chainParams.TxConfirmationTime).Return(nil, errors.New("txn receipt error"))
```

Comment:
@mankenavenkatesh : not sure why but this test case is passing whereas it should fail i think
because i am returning error from the contractCaller